projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -412,6 +411,7 @@ export class IgxOverlayService implements OnDestroy {
             document,
             true,
             info.settings.target);
+            this.addModalClasses(info);
```

Comment:
we should add modal classes right before animation started. Also we should remove modal classes right before close animation start. Please implement `removeModalClasses` method, doing the opposite of `addModalClasses` and call it before close animation starts.